ghc-app/controller.go

```diff
@@ -0,0 +1,125 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+	"gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+)
+
+// GHCController is the master controller for this application
+type GHCController struct {
+	*mux.Router
+
+	userContributions UserContributionsFunc
+	userSummary       UserSummaryFunc
+	ghcStats          GHCStatsFunc
+}
+
+// NewGHCController is the constructor for GHCController
+func NewGHCController(contributions *mgo.Collection) *GHCController {
```

Comment:
I prefer to distinguish constructors ("new") from factory methods which return a well-formed object by naming the method with `Create` rather than `New`. Very minor point, but I think it expresses intent nicely.